src/OSPSuite.Infrastructure.Import/Services/SensitivityAnalysisRunResultsImportTask.cs

```diff
@@ -0,0 +1,141 @@
+               var firstFinishedTask = await Task.WhenAny(tasks);
+
+               // Remove the selected task from the list so that you don't 
+               // process it more than once.
+               tasks.Remove(firstFinishedTask);
+
+               // Await the completed task. 
+               allImportedResults.Add(await firstFinishedTask);
+               progressUpdater.IncrementProgress();
+            }
+
+            //once all results have been imported, it is time to ensure that they are consistent
+            var results = createSensitivityAnalysisRunResultsFrom(allImportedResults, simulation);
+
+            addImportedQuantityToLogForSuccessfulImport(results);
+            return results;
+         }
+      }
+
+      private void addImportedQuantityToLogForSuccessfulImport(SensitivityAnalysisRunResultsImport sensitivityAnalysisRunResultsImport)
+      {
+         if (sensitivityAnalysisRunResultsImport.Status.Is(NotificationType.Error))
+            return;
+
+         sensitivityAnalysisRunResultsImport.AddInfo(Messages.FollowingPKParameterSensitivityWereSuccessfullyImported);
+         foreach (var pkParameterSensitivity in sensitivityAnalysisRunResultsImport.SensitivityAnalysisRunResult.AllPKParameterSensitivities)
+         {
+            sensitivityAnalysisRunResultsImport.AddInfo(pkParameterSensitivity.ToString());
+         }
+      }
+
+      private SensitivityAnalysisRunResultsImport createSensitivityAnalysisRunResultsFrom(IReadOnlyList<PKParameterSensitivitiesImport> allImportedResults, IModelCoreSimulation simulation)
+      {
+         var sensitivityAnalysisRunResultsImport = new SensitivityAnalysisRunResultsImport();
+
+         //First add all available results
+         allImportedResults.Each(import => addPKParameterSensitivitiesFromSingleFile(sensitivityAnalysisRunResultsImport, import));
+
+         //now check that the defined outputs are actually available in the population simulation
+         validateImportedQuantities(sensitivityAnalysisRunResultsImport, simulation);
+
+         return sensitivityAnalysisRunResultsImport;
+      }
+
+      private void addPKParameterSensitivitiesFromSingleFile(SensitivityAnalysisRunResultsImport sensitivityAnalysisRunResultsImport, PKParameterSensitivitiesImport pkParameterSensitivitiesImport)
+      {
+         var sensitivityAnalysisRunResult = sensitivityAnalysisRunResultsImport.SensitivityAnalysisRunResult;
+         sensitivityAnalysisRunResultsImport.PKParameterSensitivitiesImportFiles.Add(pkParameterSensitivitiesImport.PKParameterSensitivitiesImportFile);
+
+         foreach (var pkParameterSensitivity in pkParameterSensitivitiesImport.PKParameterSensitivities)
+         {
+            if (sensitivityAnalysisRunResult.HasPKParameterSensitivityWithId(pkParameterSensitivity.Id))
+               sensitivityAnalysisRunResultsImport.AddError(Error.DuplicatedPKParameterSensitivityFor(pkParameterSensitivity.Id));
```

Comment:
We error here if there is two entries for the smae QuantityPath, PKParameter, Parameter combo. This should never happen